49 document api grid (#111)
NCAR/tuv-x: docs/source/api/grid.rst

.. TUV-x Grids

Grids
=====

> A :f:type:`~tuvx_grid/grid_t` object defines the vertical extent of the model. 

Reviewer: ```suggestion
A :f:type:`~tuvx_grid/grid_t` object defines discreet values along an axis (e.g., height, wavelength). 
```

There are three 
classes that can be used to configure TUV-x. Typically, the type of grid used

> at runtime is configurd from the configuration file. See 

Reviewer: ```suggestion
at runtime is configured from the configuration file. See 
```

:ref:`configuration-grid` for more information.

All grid configurations are generated by the factory pattern inside the
:f:mod:`tuvx_grid_factory`. Each grid is 
created once and lives in the :f:type:`~tuvx_grid_warehouse/grid_warehouse_t` 
class. A copy of the grid is returned each time the grid is needed. Each class

> the depends on a grid will take a 

Reviewer: ```suggestion
that depends on a grid will take a 
```

:f:type:`~tuvx_grid_warehouse/grid_warehouse_t` object as an
argument and pull the type of grid that is desired out of the warehosue, 
like this: ::

  function run( grid_warehouse ) result( thing )
      use tuvx_grid,           only : grid_t
      use tuvx_grid_warehouse, only : grid_warehouse_t

      integer                               :: thing
      type(grid_warehouse_t), intent(inout) :: grid_warehouse
      class(grid_t),          pointer       :: zGrid => null( )

      zGrid => grid_warehouse%get_grid( "height", "km" )
      ...
    end function run

> .. f:automodule:: tuvx_grid

Reviewer: Not sure if this is fixable, but it seems like the formatting of the module description works where the module is described:
![image](https://user-images.githubusercontent.com/13121557/182235083-2ef439d4-5e2f-421f-9def-e839862cb969.png)

but when the module is used in another module/function, the formatting symbols end up in the text:
![image](https://user-images.githubusercontent.com/13121557/182235314-32549475-ba53-4c4e-99ba-6fcd23505e14.png)

Author: Ya, I noticed that. I couldn't find a way to fix this. The alternative is to remove the headings from the module and only have the description there. I can do that and then maybe these short descriptions would be more useful.

Author: Done.
